### PR TITLE
Edit card rtl fix

### PR DESCRIPTION
**Pull request by yosilevy**
Refactored ha-dialog RTL support so that it works even if header was not created with createCloseHeader. This is the case in edit-card dialog.

Fixed edit card dialog heading to support RTL - concatenation is reversed (should have been a string with a token in the localization but not worth breaking all translations for this. Maybe we want to refactor to another function there?

### CODE REVIEW

**Review by bramkragten — APPROVED** [review]

File: src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts [hunk 1]
```diff
@@ -100,9 +104,16 @@ export class HuiDialogEditCard extends LitElement {
...
+      heading = computeRTL(this.hass)
+        ? `${this.hass!.localize("ui.panel.lovelace.editor.edit_card.header")} 
+           ${this.hass!.localize(
+             `ui.panel.lovelace.editor.card.${this._cardConfig.type}.name`
+           )}`
+        : `${this.hass!.localize(
+            `ui.panel.lovelace.editor.card.${this._cardConfig.type}.name`
+          )} ${this.hass!.localize(
+            "ui.panel.lovelace.editor.edit_card.header"
+          )}`;
```
Comment (bramkragten) [inline src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts:116]: I would prefer if we would handle this correctly and break the translations, we can just add a placeholder in all existing translations to keep the same behavior.

Comment (yosilevy) [reply src/panels/lovelace/editor/card-editor/hui-dialog-edit-card.ts]: Done

File: src/translations/en.json [hunk 2]
```diff
@@ -1898,6 +1898,7 @@
           },
           "edit_card": {
             "header": "Card Configuration",
```
Comment (bramkragten) [inline src/translations/en.json:1900]: Is this used somewhere else?

Comment (yosilevy) [reply src/translations/en.json:1900]: Yes a few lines down in the `:` expression

**Pull request merged**